aas-gui/Frontend/aas-web-gui/src/components/SubmodelPlugins/CarbonFootprint.vue

```diff
@@ -0,0 +1,170 @@
+    props: ['submodelElementData'],
+    setup() {
+        const theme = useTheme();
+        const navigationStore = useNavigationStore();
+        const aasStore = useAASStore();
+
+        return {
+            theme, // Theme Object
+            navigationStore, // NavigationStore Object
+            aasStore, // AASStore Object
+        };
+    },
+    data() {
+        return {
+            PCFCO2eq: '' as string,
+            TCFCO2eq: '' as string,
+        };
+    },
+    mounted() {
+        this.initializeProductCarbonFootprint();
+        this.initializeTransportCarbonFootprint();
+    },
+    computed: {
+        SelectedNode() {
+            return this.aasStore.getSelectedNode;
+        },
+        colorPercentage() {
+            return this.calculatePercentage(this.PCFCO2eq, this.TCFCO2eq);
+        },
+        pcfValueColor() {
+            return this.getThemeColor('pcf');
+        },
+        tcfValueColor() {
+            return this.getThemeColor('tcf');
+        },
+        fingersColor() {
+            return this.getThemeColor('fingers');
+        }
+    },
+    methods: {
+        calculatePercentage(PCFCO2eq: string, TCFCO2eq: string): number {
+            const productValue = parseFloat(PCFCO2eq) || 0;
+            const transportValue = parseFloat(TCFCO2eq) || 0;
+            const total = productValue + transportValue;
+            return total ? (productValue / total) * 100 : 0;
+        },
+        getThemeColor(colorKey: string): string {
+            const isDark = this.$vuetify.theme.global.current.dark;
+            const themeColors = isDark ? this.$vuetify.theme.themes.dark.colors : this.$vuetify.theme.themes.light.colors;
+            return themeColors[colorKey];
+        },
+        initializeCarbonFootprint(semanticId: string, value: 'PCFCO2eq' | 'TCFCO2eq',) {
+            // console.log('Component Mounted. SubmodelElementData:', this.submodelElementData);
+            if (Object.keys(this.submodelElementData).length === 0) {
+                return;
+            }
+            try {
+                const carbonFootprintData = { ...this.submodelElementData };
+                const entries = carbonFootprintData.submodelElements.filter((elem: any) => {
+                    return elem.semanticId?.type === 'ExternalReference' &&
+                        elem.semanticId?.keys?.[0]?.type === 'GlobalReference' &&
+                        elem.semanticId?.keys?.[0]?.value === semanticId;
+                });
+                if (entries.length === 0) {
+                    console.warn(`No entries found for ${semanticId}.`);
+                    return;
+                }
+                const totalValue = entries.reduce((accumulator: number, entry: any) => {
+                    const co2eqValue = parseFloat(entry.value?.find((val: any) => val?.idShort?.includes('CO2eq'))?.value) || 0;
+                    return accumulator + co2eqValue;
+                }, 0);
```

Comment:
Please also return the entire property here (for each of the two types -> PCF/TCF) and not just the totalValue of the PCF values combined.
We need the entire property in the template to calculate the unit suffix as well (see comment for unit suffix).